libraries/goclient/goLibrary.go

```diff
@@ -0,0 +1,126 @@
+package goclient
+
+import (
+	"fmt"
+	"os"
+	"strconv"
+	"time"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/constants"
+
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/host"
+	"github.com/intel/afxdp-plugins-for-kubernetes/internal/uds"
+)
+
+var (
+	hWR          uds.Handler
+	hPod         host.Handler
+	busy_timeout int = 20
+	busy_budget  int = 64
+)
+
+const (
+	requestDelay = 100 * time.Millisecond
+)
+
+/*
+Returns the version of our Handshake as a string
+*/
+func GetVersionStr() string {
+	return constants.Uds.Handshake.Version
+}
+
+/*
+Call this function to initialize the library, returns a cleanup function
+*/
+func Init() uds.CleanupFunc {
+	hWR = uds.NewHandler()
+	hPod = host.NewHandler()
+
+	// init uds Handler for reading and writing
+	if err := hWR.Init(constants.Uds.PodPath, constants.Uds.Protocol, constants.Uds.MsgBufSize, constants.Uds.CtlBufSize, 0*time.Second, ""); err != nil {
+		println("Library Error: Error Initialising UDS server: ", err)
+		os.Exit(1)
+	}
+
+	cleanup, err := hWR.Dial()
+	if err != nil {
+		println("Library Error: UDS Dial error:: ", err)
+		cleanup()
+		os.Exit(1)
+	}
+	hostname, err := hPod.Hostname()
+	if err != nil {
+		println("Library Error: Failed to initialize host: ", err)
+		os.Exit(1)
+	}
+	makeRequest("/connect, "+hostname, -1)
+	return cleanup
+}
+
+/*
+Returns the version of our Handshake
+*/
+func GetVersion() {
```

Comment:
So this function should actually returns something, the version (and an error)

That would mean that your makeRequest() would need updating, I'll add a comment there, it might make sense to remove it...